src/read/dwarf.rs

```diff
@@ -1046,7 +1046,7 @@ impl<R: Reader> Unit<R> {
     /// The returned value is relative to this unit's `comp_dir`.
     pub fn dwo_name(&self) -> Result<Option<AttributeValue<R>>> {
         let mut entries = self.entries();
-        if let None = entries.next_entry()? {
+        if entries.next_entry()?.is_none() {
```

Comment:
I think I prefer:
```rust
        entries.next_entry()?;
        let entry = entries.current().ok_or(Error::MissingUnitDie)?;
```
This is more than a clippy fix, but it matches how we handle this case elsewhere in the file.